core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -204,6 +204,14 @@ static uint32_t desc_to_mattr(uint64_t desc)
 		return 0;
 	}
 
+	if (level == 3) {
+		if ((desc & L3_BLOCK_DESC) != L3_BLOCK_DESC)
```

Comment:
minor: it's a bit weird to use `L3_BLOCK_DESC` as both mask and value. same below.
(note that this was already done in `core_mmu_find_table()`). 
An `DESC_ENTRY_TYPE_MASK` (or so) could be more... explicit.